### ISSUE TITLE
img-redundant-alt warning cannot handle optional chaining operator

### ISSUE THREAD

**Issue by tehlordvortex** [jsx-eslint/eslint-plugin-jsx-a11y#794]
Hello.

I encountered an odd situation recently with `jsx-a11y/img-redundant-alt`, whereby I had an `img` tag like so:

```js
<img src={imageSource?.url} alt={imgSource?.name} />
```

And this triggered the warning. If I remove the `?.` from the value of `alt` and use just `.`, the warning goes away. That is:

```js
<img src={imageSource?.url} alt={imgSource.name} />
```

The above code snippet doesn't trigger the warning.

**Comment by ljharb** [comment]
Can you confirm that you have the latest version of `jsx-ast-utils` installed?

**Comment by tehlordvortex** [comment]
Hello,

I ran `npm ls jsx-ast-utils` and this was the output:
```
└─┬ react-scripts@4.0.3
  ├─┬ eslint-plugin-jsx-a11y@6.4.1
  │ └── jsx-ast-utils@3.2.0 
  └─┬ eslint-plugin-react@7.23.2
    └── jsx-ast-utils@3.2.0  deduped
```

According to npm: https://www.npmjs.com/package/jsx-ast-utils, this is the latest version of the package (3.2.0)

For context, this is a project bootstrapped with Create React App.

**Comment by frontendj** [comment]
The same issue happens with aria-hidden={foo?.bar}. It looks eslint doesn't like optional chaining here.
An error that is shown for aria-hidden:

```
 FAIL  __tests__/src/rules/aria-proptypes-test.js
  ● aria-proptypes › valid › <div aria-hidden={foo?.bar} />

    assert(received)

    Expected value to be equal to:
      true
    Received:
      false

    Message:
      A fatal parsing error occurred: Parsing error: Unexpected token .

      at runRuleForItem (node_modules/eslint/lib/rule-tester/rule-tester.js:563:13)
      at testValidTemplate (node_modules/eslint/lib/rule-tester/rule-tester.js:609:28)
      at Object.<anonymous> (node_modules/eslint/lib/rule-tester/rule-tester.js:889:25)
```
